tests/validation/tests/v3_api/test_global_registry.py

```diff
@@ -0,0 +1,124 @@
+REGISTRY_TEMP_ID = "catalog://?catalog=system-library&template=harbor" \
+                   "&version=1.7.5-rancher1"
+
+
+def test_global_registry_enable():
+    admin_client = setup["admin_client"]
+    p_client = setup["p_client"]
+    p = setup["project"]
+
+    setting = admin_client.by_id_setting(id="global-registry-enabled")
+    assert setting is not None, \
+        "Expected to find global-registry-enabled setting"
+
+    p_client.create_app(
+        name="global-registry",
+        externalId=REGISTRY_TEMP_ID,
+        targetNamespace="cattle-system",
+        projectId=p.id,
+        answers={
+            "expose.ingress.host": setup["server_host"],
+            "externalURL": setup["server_url"],
+            "persistence.enabled": "false",
+            "clair.enabled": "false",
+            "notary.enabled": "false"
+        },
+    )
+    admin_client.update_by_id_setting(id="global-registry-enabled",
+                                      value="true")
+    wait_for_app_to_active(p_client, "global-registry", timeout=360)
+    check_global_registry()
+
+
+@pytest.fixture(scope='module', autouse="True")
+def create_clients():
+    admin_client = get_admin_client()
+    setup["admin_client"] = admin_client
+
+    local_cluster = admin_client.by_id_cluster("local")
+    if local_cluster is None:
+        pytest.skip("skipping HA-only tests", allow_module_level=True)
+
+    projects = admin_client.list_project(
+        clusterId="local")
+    sys_p = None
+    for project in projects:
+        if project['name'] == "System":
+            sys_p = project
+            break
+    assert sys_p is not None
+    p_client = get_project_client_for_token(sys_p, ADMIN_TOKEN)
+    setup["project"] = sys_p
+    setup["p_client"] = p_client
+
+    serverURLSetting = admin_client.by_id_setting(id="server-url")
+    assert serverURLSetting is not None
+    setup["server_url"] = serverURLSetting.value
+    setup["server_host"] = setup["server_url"].strip("https://")
+
+
+def check_global_registry():
+    response = requests.get(setup["server_url"] + "/registry/",
+                            verify=False)
+    assert response.status_code == 200
+    if docker_exist() and prepare_ca_crt():
+        server_host = setup["server_host"]
+        command = '''
+        set -e;
+        docker login -u admin -p Harbor12345 %s
+        docker pull nginx
+        docker tag nginx %s/library/nginx
+        docker push %s/library/nginx
+        ''' % (server_host, server_host, server_host)
+        proc = subprocess.run(command, shell=True, universal_newlines=True)
+        assert proc.returncode == 0, proc.stderr
+    else:
+        response = requests.get(setup["server_url"] + "/v2/",
+                                verify=False)
+        assert response.status_code == 401
+        assert response.headers[DISTRIBUTION_HEADER] == 'registry/2.0'
+
+
+def docker_exist():
+    return find_executable("docker") is not None
```

Comment:
If we expect Docker to be present in the client from where the test cases are being executed , we should get this installed in the test container.